### PR TITLE
feature: Show scaladocs for signature help

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we wouldn't show any documentation for signature help. Now, we properly look for it the same way as in hover and completions

I also extracted signature help from the ScalaPresentationCompiler class.

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/SignatureHelpProvider.scala [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+      }
+    new l.SignatureHelp(
+      signatureInfos.map(signatureToSignatureInformation).asJava,
+      callableN,
+      paramN
+    )
+  end signatureHelp
+
+  private def withDocumentation(
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/SignatureHelpProvider.scala:67]: This is the only new method added along with the above:
```scala
val signatureInfos =
      alternatives.flatMap { denot =>
        val doc = search.symbolDocumentation(denot.symbol)
        (doc, Signatures.toSignature(denot)) match
          case (Some(info), Some(signature)) =>
            withDocumentation(info, signature)
          case (_, sig) => sig
```

File: mtags/src/main/scala-3/scala/meta/internal/pc/SemanticdbSymbols.scala [hunk 2]
```diff
@@ -103,7 +103,11 @@ object SemanticdbSymbols:
           decls0 ++ sym.owner.companionClass.info.decls.lookupAll(sym.name)
         else decls0
       end decls
-      val alts = decls.filter(_.isOneOf(Method | Mutable)).toList.reverse
+      // private java constructors do not have a symbol created
+      val alts = decls
+        .filter(d => d.isOneOf(Method | Mutable) && !d.is(Private))
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/SemanticdbSymbols.scala:108]: This is how it's done in the Javadoc indexer and in semanticdb for Scala 2.

File: tests/cross/src/test/scala/tests/pc/SignatureHelpDocSuite.scala [hunk 3]
```diff
@@ -125,6 +131,16 @@ class SignatureHelpDocSuite extends BaseSignatureHelpSuite {
             |        ^^^^^^^^^^^^^
             |  @param ifEmpty String the expression to evaluate if empty.
             |  @param f the function to apply if nonempty.
+            |""".stripMargin,
+      "3" ->
+        s"""|$foldLatestDocs
+            |**Parameters**
+            |- `f`: the function to apply if nonempty.
+            |- `ifEmpty`: the expression to evaluate if empty.
+            |fold[B](ifEmpty: => B)(f: A => B): B
```
Comment (dos65) [inline tests/cross/src/test/scala/tests/pc/SignatureHelpDocSuite.scala:140]: It seems that we need  `seenFrom` as in [HoverProvider](https://github.com/scalameta/metals/blob/806786093d6f3cf8f9e4aa490dbe3a94662f1230/mtags/src/main/scala-3/scala/meta/internal/pc/HoverProvider.scala#L140) but we can do it later

**Review by dos65 — APPROVED** [review]
LGTM

File: tests/cross/src/test/scala/tests/pc/SignatureHelpDocSuite.scala (hunk 3 above)
Comment (tgodzik) [reply tests/cross/src/test/scala/tests/pc/SignatureHelpDocSuite.scala:140]: I will do an update with that!

Comment (tgodzik) [reply tests/cross/src/test/scala/tests/pc/SignatureHelpDocSuite.scala:140]: Follow up here: https://github.com/scalameta/metals/pull/3941

**Pull request merged**